spec/lib/zendesk_apps_tools/api_connection_spec.rb

```diff
@@ -134,4 +136,33 @@ def initialize(subdomain = nil, username = nil)
       end
     end
   end
+
+  describe '#get_connection' do
+    context 'with missing credentials' do
+      it 'calls prepare_api_auth for password when password is missing' do
+        subject = subject_class.new('subdomain', 'username')
+        expect(subject).to receive(:prepare_api_auth).exactly(1).times
+        subject.get_connection
+      end
+
+      it 'calls prepare_api_auth for username when username is missing' do
+        subject = subject_class.new('subdomain')
+        expect(subject).to receive(:prepare_api_auth).exactly(1).times
+        subject.get_connection
+      end
+    end
+
+    context 'with all credentials' do
+      let(:subject) { subject_class.new('subdomain', 'username', 'password') }
+
+      it 'does not call prepare_api_auth for credentials' do
+        expect(subject).to receive(:prepare_api_auth).exactly(0).times
```

Comment:
`.to_not receive(prepare_api_auth)` if it works?

Reply:
Yes it does! :)